css-highlight-api-1/Overview.bs

```diff
@@ -225,25 +214,60 @@ Registering Custom Highlights</h3>
 
 	[Exposed=Window]
 	interface HighlightRegistry {
-		setlike<Highlight>;
+		maplike<DOMString, Highlight>;
 	};
 	</xmp>
 
 	<div algorithm="to register a custom highlight">
 		To [=register=] a [=custom highlight=],
-		invoke the {{HighlightRegistry/add()}} method of the [=highlight registry=]
-		with the [=custom highlight=] as the argument.
-
-		When invoked,
-		the <dfn method for=HighlightRegistry>add(Highlight value)</dfn> method must run these steps:
-
-		1. If there is already a [=set entry=] with the same {{Highlight/name}} as the {{Highlight/name}} of {{value}},
-			then [=throw=] an {{"OperationError"}}.
-		3. Let |valueArg| be the result of [=converted to an ECMAScript value|converting=] {{value}} to an ECMAScript value.
-		4. Let |result| be the result of running [[webidl#es-add-delete|the steps for a built-in setlike add function]],
-			with the [=context object=] as the <code>this</code> value
-			and with |valueArg| as the argument.
-		5. Return |result|.
+		invoke the <code>set</code> method of the [=highlight registry=]
+		which will run [[webidl#es-map-set|the steps for a built-in maplike set function]],
+		with the [=context object=] as the <code>this</code> value,
+		the passed-in [=custom highlight name=] as |keyArg|,
+		and the passed-in highlight as |valueArg|.
+
+		The <dfn>custom highlight name</dfn> assigned to a [=custom highlight=] when it is [=registered=]
+		is used to identify the highlight during styling (see [[#styling-highlights]]).
+
+	Note: When registering a [=custom highlight=], authors are recommended to use a
+	[=custom highlight name=] that is a valid CSS [=identifier=]. Using a name that is not a valid
+	identifier can make the highlight hard, and in some cases impossible, to style via CSS.
+
+	Note: It is possible to [=register=] a [=custom highlight=] with more than one [=custom highlight name=].
+	However, using more than one name to style a highlight will assign the highlight multiple different sets
+	of styles, without a way to control the stacking order of conflicting styles within these sets
+	during [[#painting|painting]]. This may be limiting for authors and may cause confusing painting behavior
+	(see the [[#styling-problems-with-multiple-names-per-highlight|example]] below for more context). Therefore,
+	<b>it is recommended that authors only use one name per highlight during styling</b>.
+	<div class=example id=styling-problems-with-multiple-names-per-highlight>
+		<xmp highlight=html>
+			<style>
+			div::highlight(foo) {
```

Comment:
Do we need this first rule? Don't the last two rules demonstrate the problem even without this?

Reply:
Yes, they do. Thanks! Removing the redundant rule.